src/votekit/election_types.py

```diff
@@ -1,95 +1,206 @@
+#             elected=self.elected,
+#             eliminated=self.eliminated,
+#             remaining=set(candidates),
+#             votes=fp_votes,
+#         )
+
+#     def run_election(self) -> ElectionState:
+#         """
+#         Runs complete STV election
+#         """
+#         profile = deepcopy(self.profile)
+
+#         if not self.next_round():
+#             raise ValueError(
+#                 f"Length of elected set equal to number of seats ({self.seats})"
+#             )
+
+#         while self.next_round():
+#             profile, outcome = self.run_step(profile)
+
+#         return outcome
+
+
+## Election Helper Functions
+
+
+class Borda:
+    def __init__(
+        self,
+        profile: PreferenceProfile,
+        seats: int,
+        borda_weights: Optional[list] = None,
+        standard: bool = True,
+    ):
+        self.state = ElectionState(
+            curr_round=0,
+            elected=[],
+            eliminated=[],
+            remaining=profile.get_candidates(),
+            profile=profile,
+            winner_votes={},
+            previous=None,
         )
+        self.seats = seats
+
+        self.num_cands = len(self.state.profile.get_candidates())
+
+        if borda_weights is None:
+            self.borda_weights = [i for i in range(self.num_cands, 0, -1)]
+        else:
+            self.borda_weights = borda_weights
 
-    def run_election(self) -> Outcome:
+        self.standard = standard
+
+    def run_borda_step(self):
         """
-        Runs complete STV election
+        Simulates a complete Borda election
         """
-        profile = deepcopy(self.profile)
+        borda_scores = {}  # {candidate : int borda_score}
+        candidates_ballots = {}  # {candidate : [ballots that ranked candidate at all]}
 
-        if not self.next_round():
-            raise ValueError(
-                f"Length of elected set equal to number of seats ({self.seats})"
-            )
+        # Populates dicts: candidate_rank_freq, candidates_ballots
+        for ballot in self.state.profile.get_ballots():
+            frequency = ballot.weight
+            rank = 0
+            for candidate in ballot.ranking:
+                candidate = str(candidate)
+                if candidate not in candidates_ballots:
+                    candidates_ballots[candidate] = []
+                candidates_ballots[candidate].append(
+                    ballot
+                )  # adds ballot where candidate was ranked
 
-        while self.next_round():
-            profile, outcome = self.run_step(profile)
+                # adds Borda score to candidate
+                if candidate not in borda_scores:
+                    borda_scores[candidate] = 0
+                if (rank + 1) <= len(self.borda_weights):
+                    borda_scores[candidate] += frequency * self.borda_weights[rank]
 
-        return outcome
+                rank += 1
 
+            if self.standard is True:
```

Comment:
what's this boolean doing, if standard is false is some different type of borda scoring supposed to occur